pkg/kubelet/kubelet.go

```diff
@@ -462,7 +463,19 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 		}
 	}
 	httpClient := &http.Client{}
-	parsedNodeIP := net.ParseIP(nodeIP)
+
+	var nodeIPs []net.IP
+	for _, nodeIP := range strings.Split(nodeIP, ",") {
+		parsedNodeIP := net.ParseIP(strings.TrimSpace(nodeIP))
+		if parsedNodeIP == nil {
+			klog.Warningf("Could not parse --node-ip value %q; ignoring", nodeIP)
+		} else {
+			nodeIPs = append(nodeIPs, parsedNodeIP)
+		}
+	}
+	if len(nodeIPs) > 2 || (len(nodeIPs) == 2 && utilnet.IsIPv6(nodeIPs[0]) == utilnet.IsIPv6(nodeIPs[1])) {
```

Comment:
Does this need to be gated?

Reply:
Define "need"...

There is no chance that a user is going to be accidentally injured by us not gating it, since they can only end up with two IPs here if they explicitly passed two IPs on the command line.

Note that the eventual side effect of setting dual-stack `podIPs` _is_ gated.